Userland/Utilities/ls.cpp

```diff
@@ -197,20 +198,38 @@ ErrorOr<int> serenity_main(Main::Arguments arguments)
 
 static int print_escaped(StringView name)
 {
-    int printed = 0;
+    auto const escaped_string = [name]() -> String {
+        bool general_escape = false;
+        bool contains_single_quotes = false;
+        bool contains_double_quotes = false;
+        for (auto const c : name) {
+            if (Shell::special_character_escape_mode(c) != Shell::SpecialCharacterEscapeMode::Untouched and c != '\'' and c != '"')
+                general_escape = true;
+            if (c == '\'')
+                contains_single_quotes = true;
+            if (c == '"')
+                contains_double_quotes = true;
+        }
+        if (contains_double_quotes and not contains_single_quotes)
+            return Shell::escape_token_for_single_quotes(name);
+        if (general_escape or contains_single_quotes or contains_double_quotes)
+            return Shell::escape_token_for_double_quotes(name);
```

Comment:
In the serenity codebase, we generally use `||`, `&&` and `!` instead of `or`, `and` and `not`.

Reply:
Someone already told me that, but I didn't think to modify this PR.
Anyway, it's modified!